### PR TITLE
microoptimizations

### PR DESCRIPTION
**Pull request by jsteemann**

### CODE REVIEW

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3629/

File: arangod/RocksDBEngine/RocksDBCollection.cpp [hunk 1]
```diff
@@ -1272,6 +1274,22 @@ Result RocksDBCollection::removeDocument(arangodb::transaction::Methods* trx,
 
   // disable indexing in this transaction if we are allowed to
   IndexingDisabler disabler(mthds, trx->isSingleOperationTransaction());
+
```
Comment (graetzer) [inline arangod/RocksDBEngine/RocksDBCollection.cpp]: shouldn't we check whether this is actually the error first ?

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3634/

File: arangod/RocksDBEngine/RocksDBCollection.cpp (hunk 1 above)
Comment (jsteemann) [reply arangod/RocksDBEngine/RocksDBCollection.cpp]: Done

**Pull request merged**